src/docgen/schema.ts

```diff
@@ -0,0 +1,428 @@
+/**
+ * Describes any kind of type. This could be a primitive, a user-defined type
+ * (like `Bucket`), or a composition of types (like `Map<string, Bucket>[]`).
+ */
+export interface TypeSchema {
+  /**
+   * The language-specific name of the type. May contain "%" placeholder
+   * values to indicate references to types defined in the "types" field.
+   *
+   * @example "string"
+   * @example "%"
+   * @example "typing.List[%]"
+   * @example "Map<%, %>"
+   */
+  readonly name: string;
```

Comment:
`name` doesn't feel like a great name if this is more like a `formattingPattern`, especially since for simple jsii types it seems the name is never in here.

Reply:
Updated